src/families/elrond/js-buildSubAccounts.ts

```diff
@@ -0,0 +1,113 @@
+import { CryptoCurrency, findTokenById, listTokensForCryptoCurrency, TokenCurrency } from "@ledgerhq/cryptoassets";
+import BigNumber from "bignumber.js";
+import { emptyHistoryCache } from "../../account";
+import { Account, SyncConfig, TokenAccount } from "../../types";
+import { getAccountESDTOperations, getAccountESDTTokens } from "./api";
+
+async function buildElrondESDTTokenAccount({
+  parentAccountId,
+  accountAddress,
+  token,
+  balance,
+}: {
+  parentAccountId: string;
+  accountAddress: string;
+  token: TokenCurrency;
+  balance: BigNumber;
+}) {
+  const extractedId = token.id;
+  const id = parentAccountId + "+" + extractedId;
+  const tokenIdentifierHex = token.id.split('/')[2];
+  const tokenIdentifier = Buffer.from(tokenIdentifierHex, 'hex').toString();
+
+  const operations = await getAccountESDTOperations(parentAccountId, accountAddress, tokenIdentifier);
+ 
+  const tokenAccount: TokenAccount = {
+    type: "TokenAccount",
+    id,
+    parentId: parentAccountId,
+    starred: false,
+    token,
+    operationsCount: operations.length,
+    operations,
+    pendingOperations: [],
+    balance,
+    spendableBalance: balance,
+    swapHistory: [],
+    creationDate:
+      operations.length > 0
+        ? operations[operations.length - 1].date : new Date(),
+    balanceHistoryCache: emptyHistoryCache, // calculated in the jsHelpers
+  };
+  return tokenAccount;
+}
+
+async function elrondBuildESDTTokenAccounts({
```

Comment:
I am not sure if this function do this yet, but what would be good is that, if the synchronisation do not find any changes in the sub token accounts, that the object reference is kept. basically if I do two sync in a row, it should often be `oldAccount.subAccounts === newAccount.subAccounts`

we managed to solve this in Ethereum here https://github.com/LedgerHQ/ledger-live-common/blob/master/src/families/ethereum/synchronisation.ts#L112-L157 and https://github.com/LedgerHQ/ledger-live-common/blob/master/src/families/ethereum/synchronisation.ts#L631

unfortunately it's not part of a "generalized framework" but feel free to grab some code from there still.

Reply:
Currently, we do not verify if there are any changes between two sync. I will add this part.